matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/SyncResponseHandler.kt

```diff
@@ -165,12 +170,28 @@ internal class SyncResponseHandler @Inject constructor(
         Timber.v("On sync completed")
         cryptoSyncHandler.onSyncCompleted(syncResponse)
 
+        monarchy.awaitTransaction {
+            trackUserSpaceNumber(it)
+        }
```

Comment:
Is it worth doing that treatment on **every** sync response?

Reply:
I've tried to do that, but I haven't found a way to implement it in a better way. I thought to only do that when there are some rooms to join in Sync, but in that case we won't send initial count before user will join any room. And we can't access here any type of app-level cache, like SharedPreferences to validate if we've send it already, since this code is in sdk module. So to do that we will require to implement some interface and then provide implementation from app module, which looks overcomplicated for this. I thought that number of rooms and spaces is relatively small and so this counting should not take a long tome.

Reply:
OK
@ganfra what do you think?

Reply:
Yes I don't think this should be done in the SDK at all if it's only for Analytics...
Maybe you can listen in the App a combination of SyncState and RoomSummary to know when it has changed?

Reply:
+1 for avoiding the concept of tracking within the SDK, it feels like we should be able to register event listeners and track from the element side